### PR TITLE
Migrate `blogStickers` to use ReactQuery 

### PR DESCRIPTION
**Pull request by flootr**
#### Changes proposed in this Pull Request

Motivated by #63350 I took the opportunity to migrate blog stickers to React Query and clean up any redux related code.

* Migrate blog stickers to use ReactQuery
  * Create and use `useBlogStickersQuery`
  * Create and use `use(Add|Remove)BlogStickersMutation`
  * Remove any Redux and data layer related code

#### Testing instructions

* Follow the testing instructions from #63350, they should still work exactly as described
* Now choose a blog you own which has at least 1 post (you should be on `/read/feeds/:siteId`)
* Click on the three-dots menu for a post and apply the `a8c-test-blog` sticker, then remove it again after you received a success message. You can also test the error case by blocking requests to `public-api.wordpress.com/rest/v1.1/sites/*/blog-stickers`.

### CODE REVIEW

File: client/blocks/blog-stickers/use-blog-stickers-query.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { useQuery } from 'react-query';
+import wp from 'calypso/lib/wp';
+
+export const useBlogStickersQuery = ( blogId, queryOptions = {} ) =>
+	useQuery( [ 'blog-stickers', blogId ], () => wp.req.get( `/sites/${ blogId }/blog-stickers` ), {
+		...queryOptions,
+		enabled: !! blogId,
+		staleTime: 1000 * 60 * 5, // 5 minutes
+		meta: {
+			persist: false,
```
Comment (tyxla) [inline client/blocks/blog-stickers/use-blog-stickers-query.js]: Any good reason not to persist them? Seems like they don't change that often, and if we re-query them on every page load, they'll mostly be fresh anyway?

File: client/blocks/reader-post-options-menu/blog-stickers.jsx [hunk 2]
```diff
@@ -1,38 +1,29 @@
...
-import getBlogStickers from 'calypso/state/selectors/get-blog-stickers';
+import { useBlogStickersQuery } from 'calypso/blocks/blog-stickers/use-blog-stickers-query';
 import ReaderPostOptionsMenuBlogStickerMenuItem from './blog-sticker-menu-item';
 
-class ReaderPostOptionsMenuBlogStickers extends Component {
-	static propTypes = {
-		blogId: PropTypes.number.isRequired,
-	};
+function ReaderPostOptionsMenuBlogStickers( { blogId } ) {
+	const blogStickersOffered = [ 'dont-recommend', 'broken-in-reader', 'a8c-test-blog' ];
```
Comment (tyxla) [inline client/blocks/reader-post-options-menu/blog-stickers.jsx]: Could be declared outside of the component.

**Review by tyxla — APPROVED** [review]
Tests great and the code looks awesome! I've only left a couple of minor suggestions.

Thank you for taking this a step further 🚀

File: client/blocks/blog-stickers/use-blog-stickers-query.js (hunk 1 above)
Comment (flootr) [reply client/blocks/blog-stickers/use-blog-stickers-query.js]: Added persistence back in 9b9f78643002c05c83f84e790adccb7275693b93 👍🏻 I was wondering whether adding / removing a sticker should result in an immediate update. Right now, I'm using a `staleTime` of 5 minutes which means the query won't be refetched on every page load if there's cached data but only if the data is older. In practice it probably won't have a big impact.

File: client/blocks/reader-post-options-menu/blog-stickers.jsx (hunk 2 above)
Comment (flootr) [reply client/blocks/reader-post-options-menu/blog-stickers.jsx]: Done via 95bbf4 👍🏻

**Comment by dsas** [comment]
@Greatdane has reached out to me on https://github.com/Automattic/wp-calypso/issues/56483#issuecomment-1131208588 to say that non-a12 users are getting a "Sorry, we had a problem retrieving block stickers. Please try again." message when visting the reader page for some blogs (see 5225778-zd-woothemes for url).

As a non-a12 user using said url, I can reproduce this issue on d6477222a2 from this issue but can't on the immediately previous commit on 1f9b52ec36.

It looks like it's because 

`client/blocks/blog-stickers/index.jsx` now fetches stickers using the API endpoint and displays that message if the query failed (which it will for non-A12s), whereas before I don't think that was the case.

I'm not sure if other changes in this merge will have had a similar effect elsewhere in Calypso.

**Comment by tyxla** [comment]
Hm, that's interesting. I looked up and it seems like the endpoint is for automatticians only.

In any case, we should suppress the error and not display it as a notice.

Ideally, I think we should enable the query only for a12s. 

What are your thoughts @flootr?

**Comment by tyxla** [comment]
@dsas here's a quick PR to make the error silent: #63808

I believe that error doesn't add a lot of value anyway.

As suggested earlier, I think a better follow-up fix would be to figure out how to disable the react query completely for non-a12s.

**Comment by tyxla** [comment]
Just filed a follow-up fix to disable the query for non-a12s in #63810.

**Pull request merged**